### ISSUE TITLE
Testing Frameworks

### ISSUE THREAD

**Issue by devinleonhart** [devinleonhart/wordsmith#15]
No pull request to master should be accepted without accompanying tests, all of which must be passing. All developers on the team need to equally contribute toward the creation of tests and should have the ability to run them.

Facebook recommends a framework called Jest to test React components. It's built off of Jasmine. However, I would like to invite you to seek alternatives and choose one you feel would be best for us if Jest is not a good match.

Choose a testing framework for use with the team and let me know what you decide. Once you have chosen, we'll need to ensure every component has its own accompanying test suite. I am looking for unit tests far more that integration tests because of React's component design promoting isolation from other components and their state.

https://facebook.github.io/react/docs/test-utils.html
https://facebook.github.io/jest/docs/tutorial-react.html

**Comment by starim** [comment]
I found [this article](https://www.toptal.com/react/how-react-components-make-ui-testing-easy) to be really informative about React testing. Highly recommend you check it out when you've got a few minutes free. It gives an overview of testing React components and also recommends using Jasmine/Karma/Webpack over Jest. [This seems to be a common sentiment.](http://kentor.me/posts/testing-react-and-flux-applications-with-karma-and-webpack/) So it looks like either Jest or a Jasmine/Karma/Webpack suite is standard. I personally don't have a favorite so I'll go with what you pick.

Also, the issue description doesn't mention this explicitly, but I would add a continuous integration system like [Travis](https://travis-ci.org/). I've done Travis integration with the Rust crates I've built and it's very helpful, so I'd like to have it in this project as well.

**Comment by devinleonhart** [comment]
I agree on Travis. I'll set that up when I get the chance.

**Comment by starim** [comment]
Stuck trying to get React to parse the JSX in tests before Karma runs them. Currently tests with JSX in them are failing with "unexpected token" because plain Javascript doesn't understand JSX like `<Footer />`.

**Comment by devinleonhart** [comment]
Same deal for me. TravisCI doesn't support the use of Webpack. This is really confusing, as it's just a commonJS module and I can verify their server is running `npm install` beforehand, and installing webpack, but every time it's the same...

```
$ webpack

/home/travis/build.sh: line 45: webpack: command not found
```

**Comment by starim** [comment]
Closed by 6d975398566fe1a0fac37449c7d890e6efc656a2
